sphinx/util/nodes.py

```diff
@@ -93,7 +93,8 @@ def findall(self, node: Node) -> Iterator[N]:
         While the `NodeMatcher` object can be used as an argument to `Node.findall`, doing so
         confounds type checkers' ability to determine the return type of the iterator.
         """
-        return node.findall(self)
+        for found in node.findall(self):
```

Comment:
Honestly, for this one, just go for a type ignore + comment saying that it's clearer that way.

Reply:
i see "ignore" and "cast" as being semantically different.

for me 'ignore' means that i have determined through runtime testing that this works, but can't or haven't written this in such a way that it's enforced/verifiable from the type system

'cast' means i can statically verify that this type is correct but that verification is opaque to mypy